ionic_user_interface/src/views/Signup.vue

```diff
@@ -0,0 +1,228 @@
+                    @keyup.enter="clickSignupButton"
+                    v-model="passwordText"
+                    name="password"
+                    type="password"
+                    required
+                  />
+                </ion-item>
+                <ion-item class="rounded">
+                  <ion-label position="stacked">Retype password</ion-label>
+                  <ion-input
+                    @keyup.enter="clickSignupButton"
+                    v-model="confirmPasswordText"
+                    name="confirmpassword"
+                    type="password"
+                    required
+                  />
+                </ion-item>
+                <ion-button
+                  id="signupButton"
+                  class="signup-button"
+                  size="medium"
+                  type="submit"
+                  expand="block"
+                >
+                  Sign up
+                </ion-button>
+                <h5>
+                  Already have an account?
+                  <router-link to="/login">Login</router-link>
+                  here!
+                </h5>
+              </form>
+            </div>
+          </ion-col>
+        </ion-row>
+      </ion-grid>
+    </ion-content>
+  </ion-page>
+</template>
+
+<script lang="ts">
+import { closeCircleOutline } from 'ionicons/icons';
+import {
+  IonButton,
+  IonCol,
+  IonContent,
+  IonGrid,
+  IonIcon,
+  IonInput,
+  IonItem,
+  IonLabel,
+  IonPage,
+  IonRow,
+} from '@ionic/vue';
+import { defineComponent, inject, ref, Ref, onBeforeUpdate } from 'vue';
+import Header from '../components/Header.vue';
+import router from '@/router';
+
+export default defineComponent({
+  name: 'About',
+  components: {
+    IonButton,
+    IonCol,
+    IonContent,
+    IonGrid,
+    IonIcon,
+    IonInput,
+    IonItem,
+    IonLabel,
+    IonPage,
+    IonRow,
+    Header,
+  },
+  setup() {
+    // Email validation regex taken from https://stackoverflow.com/questions/46155/how-to-validate-an-email-address-in-javascript
+    const emailPattern =
+      /^(([^<>()[\]\\.,;:\s@"]+(\.[^<>()[\]\\.,;:\s@"]+)*)|(".+"))@((\[[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\.[0-9]{1,3}\])|(([a-zA-Z\-0-9]+\.)+[a-zA-Z]{2,}))$/;
+    const isLoggedIn: Ref<boolean> | undefined = inject('isLoggedIn');
+    const emailText = ref('');
+    const usernameText = ref('');
+    const passwordText = ref('');
+    const confirmPasswordText = ref('');
+    const showErrorMsg = ref(false);
+    const errorMsg = ref('');
+
+    onBeforeUpdate(() => {
+      // Redirect to home if already logged in
+      if (isLoggedIn && isLoggedIn.value === true) {
+        router.push('/home');
+      }
+    });
+
+    const isValidEmail = (email: string): boolean => {
+      return emailPattern.test(email.toLowerCase());
+    };
+
+    const isValidPassword = (password: string): boolean => {
+      return password.length >= 6;
+    };
+
+    const isValidUsername = (username: string): boolean => {
```

Comment:
Lets limit to alphanumeric and space so won't be too funky

https://github.com/yarkhinephyo/yolo_bouldering/blob/f2f0b9f5b1afc41fcbb8a875c8b0f5522282eff8/lambda_backend/user_microservice/src/common/schema.ts#L1